torch_xla/amp/autocast_mode.py

```diff
@@ -1,5 +1,40 @@
 import torch
+import torch_xla.core.xla_model as xm
+from typing import Any
 
-autocast = torch.cuda.amp.autocast
-custom_fwd = torch.cuda.amp.custom_fwd
-custom_bwd = torch.cuda.amp.custom_bwd
+
+class autocast(torch.amp.autocast_mode.autocast):
+  r"""
+    See :class:`torch.autocast`.
+    ``torch_xla.amp.autocast(device, args...)`` is equivalent to ``torch.autocast("xla", args...)`` for TPUs
+    ``torch.autocast("cuda", args...)`` for GPUs.
+    """
+
+  def __init__(self,
+               device,
+               enabled: bool = True,
+               dtype: torch.dtype = torch.bfloat16,
+               cache_enabled: bool = True):
+    if xm.xla_device_hw(device) == 'GPU':
```

Comment:
Is fp16 only available on TPU?

Reply:
So is it hardware-dependent whether bf16 is supported? XLA itself does not impose any restrictions, so why is there a need for this judgment here?

Reply:
I think we can relax the restriction here as a follow up, through I don't know what's a good way to tell whether current GPU handles bf16 or not. @cowanmeg FYI

Reply:
Yes, we can relax the dtype for GPUs. Since this is a wrapper around PyTorch's autocast class, we can use their checks to see if the GPU handles bf16.
https://github.com/pytorch/pytorch/blob/main/torch/amp/autocast_mode.py#L269